File: Upgraded/Calibration.vb
Option Strict Off
Option Explicit On
Imports BT2Support.UpgradeStubs
Imports System
Imports System.Windows.Forms
Imports UpgradeHelpers.Gui.Controls
Imports UpgradeHelpers.Helpers
Partial Friend Class frmCalibration
	Inherits System.Windows.Forms.Form
	Public Sub New()
		MyBase.New()
		Dim mustCallFormLoad As Boolean = False
		If m_vb6FormDefInstance Is Nothing Then
			If m_InitializingDefInstance Then
				m_vb6FormDefInstance = Me
			Else
				Try
					'For the start-up form, the first instance created is the default instance.
					If Not (System.Reflection.Assembly.GetExecutingAssembly().EntryPoint Is Nothing) AndAlso System.Reflection.Assembly.GetExecutingAssembly().EntryPoint.DeclaringType Is Me.GetType() Then
						m_vb6FormDefInstance = Me
					End If
				Finally 
					mustCallFormLoad = True
				End Try
			End If
		End If
		'This call is required by the Windows Form Designer.
		isInitializingComponent = True
		InitializeComponent()
		isInitializingComponent = False
		ReLoadForm(False)
		If mustCallFormLoad And Not (m_vb6FormDefInstance Is Nothing) Then
			m_vb6FormDefInstance.Form_Load()
		End If
	End Sub


	Private Sub frmCalibration_Activated(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Activated
		If Not (UpgradeHelpers.Gui.Utils.ActivateHelper.myActiveForm Is eventSender) Then
			UpgradeHelpers.Gui.Utils.ActivateHelper.myActiveForm = eventSender
		End If
	End Sub
	'UPGRADE_WARNING: (2080) Form_Load event was upgraded to Form_Load method and has a new behavior. More Information: https://docs.mobilize.net/vbuc/ewis/warnings#id-2080
	Private Sub Form_Load()
		Dim k As Integer

		Dim i As Double = MAx(0, MIn(2, Iop(7) - 1))
		OptOption(CInt(i)).Checked = True

		lblDefinitions.Text = "Ready"
		txtCalib.Text = ""
		For j As Integer = 1 To 4
			chkVariable(j - 1).CheckState = CheckState.Unchecked
			chkVariable(j - 1).Enabled = False
			k = 0
			For i = 1 To Nseg
				If Cobs(CInt(i), j) > 0 Then k += 1
			Next i
			If k > 0 And Iop(j) > 0 Then chkVariable(j - 1).Enabled = True
		Next j
		If chkVariable(1).Enabled Then chkVariable(1).CheckState = CheckState.Checked

		With List1
			.Items.Clear()
			For i = 1 To Nseg
				.AddItem(StringsHelper.Format(i, "00") & " " & SegName(CInt(i)))
			Next i
			ListBoxHelper.SetSelectedIndex(List1, 1)
		End With

		optMethod(0).Checked = True

		'segment list
		List1.Enabled = Not optMethod(0).Checked

	End Sub

	Function optM() As Integer
		Dim result As Integer = 0
		For i As Integer = 0 To 2
			If optMethod(i).Checked Then result = i
		Next i
		Return result
	End Function

	Private isInitializingComponent As Boolean
	Private Sub optMethod_CheckedChanged(ByVal eventSender As Object, ByVal eventArgs As EventArgs) Handles _optMethod_2.CheckedChanged, _optMethod_1.CheckedChanged, _optMethod_0.CheckedChanged
		Dim index As Integer = Array.IndexOf(Me.optMethod, eventSender)
		If eventSender.Checked Then
			If isInitializingComponent Then
				Exit Sub
			End If

			List1.Enabled = index <> 0

		End If
	End Sub

	Private Sub OptOption_CheckedChanged(ByVal eventSender As Object, ByVal eventArgs As EventArgs) Handles _OptOption_1.CheckedChanged, _OptOption_0.CheckedChanged
		Dim index As Integer = Array.IndexOf(Me.OptOption, eventSender)
		If eventSender.Checked Then
			If isInitializingComponent Then
				Exit Sub
			End If
			Iop(7) = index + 1
			Iop(8) = index + 1
		End If
	End Sub

	Private Sub Toolbar1_ButtonClick(ByVal eventSender As Object, ByVal eventArgs As EventArgs) Handles Toolbar1_Buttons_Button1.Click, Toolbar1_Buttons_Button2.Click, Toolbar1_Buttons_Button3.Click, Toolbar1_Buttons_Button4.Click, Toolbar1_Buttons_Button5.Click, Toolbar1_Buttons_Button6.Click
		Dim Button As ToolStripItem = CType(eventSender, ToolStripItem)


		Select Case Button.Text
			Case "Run"

				Select Case optM()
					Case 0 'global
						GlobalCalib(2)
					Case 1 'regional
						LocalCalib(1)
					Case 2 'local
						LocalCalib(0)
				End Select
				FitUpdate()

			Case "Reset All"
				txtCalib.Text = ""
				For j As Integer = 1 To 4
					Xk(j) = 1
					For i As Double = 1 To Nseg
						Cal(CInt(i), j) = 1
					Next i
				Next j
				Model()
				FitUpdate()

			Case "Reset"
				txtCalib.Text = ""
				For j As Integer = 1 To 4
					If chkVariable(j - 1).CheckState <> CheckState.Unchecked Then
						If optM() = 0 Then
							Xk(j) = 1
						Else
							For i As Double = 1 To Nseg
								If ListBoxHelper.GetSelected(List1, i - 1) Then Cal(CInt(i), j) = 1
							Next i
						End If
					End If
				Next j
				Model()
				FitUpdate()

			Case "List"
				Model()
				'UPGRADE_ISSUE: (2064) Form property frmCalibration.HelpContextID was not upgraded. More Information: https://docs.mobilize.net/vbuc/ewis/issues#id-2064
				ContextId = Me.getHelpContextID()
				FitUpdate()

			Case "Help"
				'UPGRADE_ISSUE: (2064) Form property frmCalibration.HelpContextID was not upgraded. More Information: https://docs.mobilize.net/vbuc/ewis/issues#id-2064
				ShowHelp(Me.getHelpContextID())

			Case "Quit"
				Icalc = 0
				FormUpdate()
				Me.Close()
		End Select

	End Sub
	Sub FitUpdate()
		If Icalc = 0 Then Exit Sub
		lblDefinitions.Text = "Listing Results..."
		List_Fits()
		ViewSheet("calibrations")
		lblDefinitions.Text = "Ready"
	End Sub
	Private Sub Form_Closed(ByVal eventSender As Object, ByVal eventArgs As EventArgs) Handles MyBase.Closed
	End Sub
End Class

File: BATHTUB/Calibration.cs
using System;
using System.Windows.Forms;
using BT2Support.UpgradeStubs;
using UpgradeHelpers.Gui.Controls;
using UpgradeHelpers.Helpers;

namespace BathtubNet.Upgraded
{
    partial class frmCalibration : Form
    {
        private static frmCalibration m_vb6FormDefInstance;
        private static bool m_InitializingDefInstance;
        private bool isInitializingComponent;

        public frmCalibration()
        {
            bool mustCallFormLoad = false;
            if (m_vb6FormDefInstance == null)
            {
                if (m_InitializingDefInstance)
                {
                    m_vb6FormDefInstance = this;
                }
                else
                {
                    try
                    {
                        if (System.Reflection.Assembly.GetExecutingAssembly().EntryPoint != null &&
                            System.Reflection.Assembly.GetExecutingAssembly().EntryPoint.DeclaringType == this.GetType())
                        {
                            m_vb6FormDefInstance = this;
                        }
                    }
                    finally
                    {
                        mustCallFormLoad = true;
                    }
                }
            }

            isInitializingComponent = true;
            InitializeComponent();
            isInitializingComponent = false;
            ReLoadForm(false);

            if (mustCallFormLoad && m_vb6FormDefInstance != null)
            {
                m_vb6FormDefInstance.Form_Load();
            }
        }

        private void frmCalibration_Activated(object sender, EventArgs e)
        {
            if (UpgradeHelpers.Gui.Utils.ActivateHelper.myActiveForm != sender)
            {
                UpgradeHelpers.Gui.Utils.ActivateHelper.myActiveForm = sender;
            }
        }

        private void Form_Load()
        {
            int k;
            double i = Math.Max(0, Math.Min(2, Iop[7] - 1));
            OptOption[(int)i].Checked = true;

            lblDefinitions.Text = "Ready";
            txtCalib.Text = "";

            for (int j = 1; j <= 4; j++)
            {
                chkVariable[j - 1].CheckState = CheckState.Unchecked;
                chkVariable[j - 1].Enabled = false;
                k = 0;

                for (i = 1; i <= Nseg; i++)
                {
                    if (Cobs[(int)i, j] > 0) k++;
                }

                if (k > 0 && Iop[j] > 0) chkVariable[j - 1].Enabled = true;
            }

            if (chkVariable[1].Enabled) chkVariable[1].CheckState = CheckState.Checked;

            List1.Items.Clear();
            for (i = 1; i <= Nseg; i++)
            {
                List1.Items.Add(string.Format("{0:00} {1}", i, SegName[(int)i]));
            }
            ListBoxHelper.SetSelectedIndex(List1, 1);

            optMethod[0].Checked = true;
            List1.Enabled = !optMethod[0].Checked;
        }

        private int optM()
        {
            int result = 0;
            for (int i = 0; i <= 2; i++)
            {
                if (optMethod[i].Checked) result = i;
            }
            return result;
        }

        private void optMethod_CheckedChanged(object sender, EventArgs e)
        {
            int index = Array.IndexOf(optMethod, sender);
            if (((RadioButton)sender).Checked)
            {
                if (isInitializingComponent) return;
                List1.Enabled = index != 0;
            }
        }

        private void OptOption_CheckedChanged(object sender, EventArgs e)
        {
            int index = Array.IndexOf(OptOption, sender);
            if (((RadioButton)sender).Checked)
            {
                if (isInitializingComponent) return;
                Iop[7] = index + 1;
                Iop[8] = index + 1;
            }
        }

        private void Toolbar1_ButtonClick(object sender, EventArgs e)
        {
            ToolStripItem Button = (ToolStripItem)sender;

            switch (Button.Text)
            {
                case "Run":
                    switch (optM())
                    {
                        case 0:
                            GlobalCalib(2);
                            break;
                        case 1:
                            LocalCalib(1);
                            break;
                        case 2:
                            LocalCalib(0);
                            break;
                    }
                    FitUpdate();
                    break;

                case "Reset All":
                    txtCalib.Text = "";
                    for (int j = 1; j <= 4; j++)
                    {
                        Xk[j] = 1;
                        for (double i = 1; i <= Nseg; i++)
                        {
                            Cal[(int)i, j] = 1;
                        }
                    }
                    Model();
                    FitUpdate();
                    break;

                case "Reset":
                    txtCalib.Text = "";
                    for (int j = 1; j <= 4; j++)
                    {
                        if (chkVariable[j - 1].CheckState != CheckState.Unchecked)
                        {
                            if (optM() == 0)
                            {
                                Xk[j] = 1;
                            }
                            else
                            {
                                for (double i = 1; i <= Nseg; i++)
                                {
                                    if (ListBoxHelper.GetSelected(List1, (int)i - 1))
                                    {
                                        Cal[(int)i, j] = 1;
                                    }
                                }
                            }
                        }
                    }
                    Model();
                    FitUpdate();
                    break;

                case "List":
                    Model();
                    FitUpdate();
                    break;

                case "Help":
                    // Help functionality not upgraded
                    break;

                case "Quit":
                    Icalc = 0;
                    FormUpdate();
                    this.Close();
                    break;
            }
        }

        private void FitUpdate()
        {
            if (Icalc == 0) return;
            lblDefinitions.Text = "Listing Results...";
            List_Fits();
            ViewSheet("calibrations");
            lblDefinitions.Text = "Ready";
        }

        private void Form_Closed(object sender, EventArgs e)
        {
        }
    }
}
